### ISSUE TITLE
indent does not report MemberExpression indents in function body of variable declarations

### ISSUE THREAD

**Issue by kylehg** [eslint/eslint#7366]
**Tell us about your environment**

* **ESLint Version:** 3.7.1
* **Node Version:** 4.6.0
* **npm Version:** 2.15.9

**What parser (default, Babel-ESLint, etc.) are you using?** Default

**Please show your full configuration:**

```
{
  "rules": {
    "indent": [2, 2, {"MemberExpression": 1}]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

Running eslint with `./node_modules/.bin/eslint --fix test.js`, where `test.js` is:

```js
function Class() {}

Class.prototype.method = function () {
  return Promise.resolve(3)
      .then(function (n) {
        return n
      })
}
```

**What did you expect to happen?**

For the `.then()` block to be fixed to a 4-space indentation from its 6-space indentation

**What actually happened? Please include the actual, raw output from ESLint.**

No change. (The command executed with no errors.) Also note that running without `--fix` does not produce a lint error.

Note that I also tested with indentation in the constructor, in a normal `function`, and with ES6 methods (using `"env": {"es6": true}`), and those all work as expected, i.e., the `.then` block is correctly indented to 4 spaces.

**Comment by not-an-aardvark** [comment]
Thanks for the report. I can reproduce this issue.

It looks like `indent` is intended to not report MemberExpression access on variable declarations:

```js
var foo = bar
                                  .baz() // not reported, this is a variable declaration
```

In this case, the access isn't getting reported because the function that contains it is part of a variable declaration. This seems like a bug.

```js
/* eslint indent: [2, 2, {"MemberExpression": 1}] */
var foo = function() {
    foo
                                                           .bar;
}
```

**Comment by sstern6** [comment]
@not-an-aardvark working on this if thats ok, should have this completed by Monday morning

**Comment by sstern6** [comment]
@not-an-aardvark got it working, PR will be open tonight.
